exporter/sapmexporter/exporter.go

```diff
@@ -74,64 +83,80 @@ func newSAPMTraceExporter(cfg *Config, params component.ExporterCreateParams) (c
 		exporterhelper.WithRetry(cfg.RetrySettings),
 		exporterhelper.WithTimeout(cfg.TimeoutSettings),
 	)
-}
 
-// tracesByAccessToken takes a pdata.Traces struct and will iterate through its ResourceSpans' attributes,
-// regrouping by any SFx access token label value if Config.AccessTokenPassthrough is enabled.  It will delete any
-// set token label in any case to prevent serialization.
-// It returns a map of newly constructed pdata.Traces keyed by access token, defaulting to empty string.
-func (se *sapmExporter) tracesByAccessToken(td pdata.Traces) map[string]pdata.Traces {
-	tracesByToken := make(map[string]pdata.Traces, 1)
-	resourceSpans := td.ResourceSpans()
-	for i := 0; i < resourceSpans.Len(); i++ {
-		resourceSpan := resourceSpans.At(i)
-		accessToken := ""
-		attrs := resourceSpan.Resource().Attributes()
-		if se.config.AccessTokenPassthrough {
-			attributeValue, ok := attrs.Get(splunk.SFxAccessTokenLabel)
-			if ok {
-				accessToken = attributeValue.StringVal()
-			}
-		}
-		attrs.Delete(splunk.SFxAccessTokenLabel)
+	if err != nil {
+		return nil, err
+	}
 
-		traceForToken, ok := tracesByToken[accessToken]
-		if !ok {
-			traceForToken = pdata.NewTraces()
-			tracesByToken[accessToken] = traceForToken
+	// If AccessTokenPassthrough enabled, split the incoming Traces data by splunk.SFxAccessTokenLabel,
+	// this ensures that we get batches of data for the same token when pushing to the backend.
+	if cfg.AccessTokenPassthrough {
+		te = &baseTracesExporter{
+			Component:      te,
+			TracesConsumer: batchperresourceattr.NewBatchPerResourceTraces(splunk.SFxAccessTokenLabel, te),
 		}
-
-		// Append ResourceSpan to trace for this access token
-		traceForTokenSize := traceForToken.ResourceSpans().Len()
-		traceForToken.ResourceSpans().Resize(traceForTokenSize + 1)
-		resourceSpan.CopyTo(traceForToken.ResourceSpans().At(traceForTokenSize))
 	}
-
-	return tracesByToken
+	return te, nil
 }
 
 // pushTraceData exports traces in SAPM proto by associated SFx access token and returns number of dropped spans
 // and the last experienced error if any translation or export failed
-func (se *sapmExporter) pushTraceData(ctx context.Context, td pdata.Traces) (droppedSpansCount int, err error) {
-	traces := se.tracesByAccessToken(td)
-	droppedSpansCount = 0
-	for accessToken, trace := range traces {
-		batches, translateErr := jaeger.InternalTracesToJaegerProto(trace)
-		if translateErr != nil {
-			droppedSpansCount += trace.SpanCount()
-			err = consumererror.Permanent(translateErr)
-			continue
+func (se *sapmExporter) pushTraceData(ctx context.Context, td pdata.Traces) (int, error) {
+	rss := td.ResourceSpans()
+	if rss.Len() == 0 {
+		return 0, nil
+	}
+
+	accessToken := se.retrieveAccessToken(rss.At(0))
```

Comment:
Please add a comment to explain why the access token of element at index 0 can be used for all.

Reply:
Done.